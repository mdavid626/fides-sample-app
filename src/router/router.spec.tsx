import React from 'react';
import { cleanup, render } from '@testing-library/react';
import MoviesPage from '../pages/movies-page/movies-page';
import AboutPage from '../pages/about-page/about-page';
import FavouritesPage from '../pages/favourites-page/favourites-page';
import Router from './router';

jest.mock('../pages/movies-page/movies-page');
jest.mock('../pages/about-page/about-page');
jest.mock('../pages/favourites-page/favourites-page');

describe('router', () => {
  beforeEach(() => {
    (MoviesPage as jest.Mock).mockReturnValue(<div>movies-page</div>);
    (AboutPage as jest.Mock).mockReturnValue(<div>about-page</div>);
    (FavouritesPage as jest.Mock).mockReturnValue(<div>favourites-page</div>);
  });
  afterEach(cleanup);
  afterEach(jest.resetAllMocks);

  it('should render', () => {
    const { asFragment } = render(<Router />);
    expect(asFragment()).toMatchSnapshot();
    expect(MoviesPage).toHaveBeenCalledWith({}, {});
    expect(AboutPage).not.toHaveBeenCalled();
    expect(FavouritesPage).not.toHaveBeenCalled();
  });
});
