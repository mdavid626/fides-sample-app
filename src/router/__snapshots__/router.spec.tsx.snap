// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router should render 1`] = `
<DocumentFragment>
  <div>
    movies-page
  </div>
</DocumentFragment>
`;
